src/common/auditory_description.js

```diff
@@ -164,45 +164,556 @@ sre.AuditoryDescription.speechString = function(descrs, opt_separator) {
  * @private
  */
 sre.AuditoryDescription.toString_ = function(descrs, separator) {
-  return sre.Engine.getInstance().ssml ?
-      sre.AuditoryDescription.toSsmlString_(descrs, separator) :
-      sre.AuditoryDescription.toSimpleString_(descrs, separator);
+  switch (sre.Engine.getInstance().markup) {
+  case sre.Engine.Markup.SABLE:
+  case sre.Engine.Markup.SSML:
+  case sre.Engine.Markup.VOICEXML:
+    return sre.AuditoryDescription.toMarkupString_(descrs, separator);
+  case sre.Engine.Markup.ACSS:
+    return sre.AuditoryDescription.toAcssString_(descrs, separator);
+  case sre.Engine.Markup.NONE:
+  default:
+    return sre.AuditoryDescription.toSimpleString_(descrs, separator);
+  }
 };
 
 
-//TODO: (MOSS) WP2.3
-// Implement translations into SSML and CSS.
-// Cleaner than the hack for the NVDA/EmacsSpeak bridge!
-//
 /**
- * Translates a list of auditory descriptions into a string with SSML markup.
+ * Translates a list of auditory descriptions into a simple string.
  * @param {!Array.<sre.AuditoryDescription>} descrs The list of descriptions.
  * @param {string} separator The separator string.
- * @return {string} The generated string with SSML markup.
+ * @return {string} The generated string.
  * @private
  */
-sre.AuditoryDescription.toSsmlString_ = function(descrs, separator) {
+sre.AuditoryDescription.toSimpleString_ = function(descrs, separator) {
   return sre.BaseUtil.removeEmpty(
       descrs.map(
-      function(x) {
-        if (x.personality && x.personality.PAUSE) {
-          return '<break time="' + x.personality.PAUSE + 'ms"/>';
-        }
-        return x.descriptionString();})).
+      function(x) {return x.descriptionString();})).
       join(separator);
 };
 
 
 /**
```

Comment:
Interface for markup class: 
* markup (currenty toXYZString_) 
* error 
* join (currently mergeString_)

Abstract markup class + abstract XML markup class